### PR TITLE
FEAT: get and reset as properties of createSuite result

### PR DESCRIPTION
**Pull request by vligas**
| Q                | A                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix?         | ✖                                                                        |
| New feature?     | ✔                                                                      |
| Breaking change? | ✖                                                                        |
| Deprecations?    | ✖                                                                        |
| Documentation?   | ✖                                                                        |
| Tests added?     | ✔                                                                        |
| Fixed issues     | #315  |

Hi, I followed your advice and load the properties into the function that is beign returned by createSuite, I'm a bit insecure about the changes in the definition file since it's my first contribution to an open source project and I couldn't find any type definition for the reset and get functions. Hope that it helped somehow

### CODE REVIEW

**Comment by ealush** [comment]
Oh, you're right. I missed those types in the declarations file.
I just added them, so you can re-use.

https://github.com/ealush/vest/pull/333

File: packages/vest/src/core/createSuite/index.js [hunk 1]
```diff
@@ -53,8 +55,17 @@ const createSuite = (name, tests) => {
...
+      name: {
+        value: name,
+      },
+      get: {
+        value: () => get(name),
+      },
+      reset: {
+        value: () => reset(name),
+      },
+    }
```
Comment (ealush) [inline packages/vest/src/core/createSuite/index.js:68]: Nice!

File: packages/vest/src/vest.d.ts [hunk 2]
```diff
@@ -313,6 +313,12 @@ declare module 'vest' {
     memo(fieldName: string, testFn: TestCB, dependencies: any[]): VestTest;
   }
 
+  interface ICreateResult {
+    (...args: any[]): IVestResult;
+    get: any;
+    reset: any;
+  }
```
Comment (ealush) [inline packages/vest/src/vest.d.ts:320]: Sorry about that. You should be able to copy over the new types I created.

**Review by ealush — APPROVED** [review]
Very well done @vligas, that's exactly what I had in mind. Thank you for taking the time improving Vest.
🙏🏻

I'll merge once you handle those two tiny comments.

File: packages/vest/src/core/createSuite/spec.js [hunk 3]
```diff
@@ -94,6 +96,23 @@ describe('Test createSuite module', () => {
...
+
+    it('Should be able to reset the suite from the result of createSuite', () => {
+      const suiteId = 'test_reset_suite';
+      const testSuite = createSuite(suiteId, () => {
+        dummyTest.failing('f1', 'm1');
+      });
+      testSuite();
+      expect(get(suiteId).hasErrors()).toBe(true);
+      testSuite.reset();
+      expect(get(suiteId).hasErrors()).toBe(false);
```
Comment (ealush) [inline packages/vest/src/core/createSuite/spec.js:113]: I would also add another more explicit check here 
```suggestion
      testSuite();
      expect(get(suiteId).hasErrors()).toBe(true);
      expect(get(suiteId).testCount).toBe(1);
      testSuite.reset();
      expect(get(suiteId).hasErrors()).toBe(false);
      expect(get(suiteId).testCount).toBe(0);
```

**Comment by vligas** [comment]
Thanks for the feedback, I will add the changes asap

**Comment by ealush** [comment]
Very nice, @vligas. Thank you again for helping out 💯

**Pull request merged**